pkg/auth/handler/webapp/send_oob_otp.go

```diff
@@ -0,0 +1,127 @@
+package webapp
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/authgear/authgear-server/pkg/auth/handler/webapp/viewmodels"
+	"github.com/authgear/authgear-server/pkg/auth/webapp"
+	"github.com/authgear/authgear-server/pkg/lib/authn"
+	"github.com/authgear/authgear-server/pkg/lib/infra/mail"
+	"github.com/authgear/authgear-server/pkg/lib/interaction"
+	"github.com/authgear/authgear-server/pkg/util/httproute"
+	"github.com/authgear/authgear-server/pkg/util/phone"
+	"github.com/authgear/authgear-server/pkg/util/template"
+)
+
+const AuthenticatorIndex = 0
+
+type AuthenticationBeginNode interface {
+	GetAuthenticationEdges() ([]interaction.Edge, error)
+}
+
+var TemplateWebSendOOBOTPHTML = template.RegisterHTML(
+	"web/send_oob_otp.html",
+	components...,
+)
+
+type SendOOBOTPViewModel struct {
+	AlternativeSteps []viewmodels.AlternativeStep
+	OOBOTPTarget     string
+	OOBOTPCodeLength int
+	OOBOTPChannel    string
+}
+
+func ConfigureSendOOBOTPRoute(route httproute.Route) httproute.Route {
+	return route.
+		WithMethods("OPTIONS", "POST", "GET").
+		WithPathPattern("/send_oob_otp")
+}
+
+type SendOOBOTPHandler struct {
+	ControllerFactory ControllerFactory
+	BaseViewModel     *viewmodels.BaseViewModeler
+	Renderer          Renderer
+}
+
+type TriggerOOBOTPEdge interface {
+	GetOOBOTPTarget(idx int) string
+	GetOOBOTPChannel(idx int) string
+}
+
+func (h *SendOOBOTPHandler) GetData(r *http.Request, rw http.ResponseWriter, graph *interaction.Graph) (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+
+	baseViewModel := h.BaseViewModel.ViewModel(r, rw)
+	viewModel := EnterOOBOTPViewModel{}
+
+	var node AuthenticationBeginNode
+	if !graph.FindLastNode(&node) {
+		panic("send_oob_otp: expected graph has node implementing AuthenticationBeginNode")
+	}
+
+	edges, err := node.GetAuthenticationEdges()
+	if err != nil {
+		return nil, err
+	}
+
+	if edge, ok := edges[0].(TriggerOOBOTPEdge); ok {
+		viewModel.OOBOTPChannel = edge.GetOOBOTPChannel(AuthenticatorIndex)
```

Comment:
Can you help refactoring the logic in `AddAuthenticationAlternatives` to use the new functions?